REGRESSION: imported/w3c/web-platform-tests/navigation-api/navigation-methods/return-value/navigate-intercept.html is a flaky crash

basuke opened the pull request:
#### 030eaadb53ad6d367ea5c49014d4be49a78bddad
<pre>
REGRESSION: imported/w3c/web-platform-tests/navigation-api/navigation-methods/return-value/navigate-intercept.html is a flaky crash
<a href="https://bugs.webkit.org/show_bug.cgi?id=297793">https://bugs.webkit.org/show_bug.cgi?id=297793</a>
<a href="https://rdar.apple.com/158957951">rdar://158957951</a>

Unreviewed test gardening.

These tests are fixed in <a href="https://commits.webkit.org/299084@main">https://commits.webkit.org/299084@main</a>

* LayoutTests/TestExpectations:

Canonical link: <a href="https://commits.webkit.org/299259@main">https://commits.webkit.org/299259@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4ad31e9abdad0324cdb44451a6a7e3cf148a4c52

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/118131 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/37809 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/28445 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/124286 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/70170 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/60460e8e-fd6a-4c1d-8b35-8893ceebd74e) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/38502 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/46391 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/89643 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/59273 "") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/d81fd3b3-d332-4265-9bbc-f26b01c2a17f) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/121084 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/30665 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/105915 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/70136 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/755078e4-da38-4153-b7b6-d0fa9bdf1884) 
| | [⏳ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/iOS-18-Simulator-WPT-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/24032 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/67954 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/100091 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/24209 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/127363 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/45034 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/33942 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/98321 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/45395 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/102137 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/98107 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24996 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/43505 "Passed tests") | [⏳ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/macOS-Sonoma-Release-WK2-Intel-Tests-EWS "Waiting to run tests") | [❌ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/41508 "Hash 4ad31e9a for PR 49901 does not build (failure)") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/44906 "Built successfully") | | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/44366 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/47711 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/46055 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on LayoutTests/TestExpectations:
```diff
@@ -7086,7 +7086,6 @@ imported/w3c/web-platform-tests/html/semantics/interactive-elements/the-dialog-e
 imported/w3c/web-platform-tests/navigation-api/navigate-event/cross-origin-traversal-redirect.html [ Timeout ]
-webkit.org/b/291451 imported/w3c/web-platform-tests/navigation-api/navigation-methods/return-value/navigate-intercept-interrupted.html [ Pass Crash ]
 webkit.org/b/297477 imported/w3c/web-platform-tests/navigation-api/navigate-event/navigate-history-back-bfcache.html [ Timeout Pass Failure ]
 webkit.org/b/297793 imported/w3c/web-platform-tests/navigation-api/navigation-methods/return-value/navigate-intercept.html [ Crash Pass ]
```
Makes me wonder if this is fixed too. Seems 99% likely that it's the same issue.

basuke replied:
I don't know why I forget to save the file with second line deleted.

basuke committed:
REGRESSION: imported/w3c/web-platform-tests/navigation-api/navigation-methods/return-value/navigate-intercept.html is a flaky crash
https://bugs.webkit.org/show_bug.cgi?id=297793
rdar://158957951

Unreviewed test gardening.

These tests are fixed in https://commits.webkit.org/299084@main

* LayoutTests/TestExpectations:

Canonical link: https://commits.webkit.org/299259@main
